src/nvim/extmark.c

```diff
@@ -111,13 +111,15 @@ void extmark_set(buf_T *buf, uint32_t ns_id, uint32_t *idp, int row, colnr_T col
             // apparently incapable of handling basic integer constants.
             // This can be underanged as soon as we bump minimal gcc version.
             old_mark.flags = (uint16_t)(old_mark.flags
-                                        | (decor->hl_eol ? (uint16_t)MT_FLAG_HL_EOL : (uint16_t)0));
+                                        | (uint16_t)(decor->hl_eol ? MT_FLAG_HL_EOL : 0));
             old_mark.priority = decor->priority;
           }
+          old_mark.flags |= no_undo ? MT_FLAG_NO_UNDO : 0;
```

Comment:
I was also referring to this here https://github.com/neovim/neovim/pull/25882#discussion_r1382553709. My current tests failed without setting the flags for `old_mark`. I'll try to add a test that fails without line 137-139, I thought I had one before.

Reply:
sure yes. A test would be good. My aim is to clean up the mess that is the flags (the concept of "external" flags is a bit fuzzy and outdated), once the existing batch of open PR:s are done.